circuits/batch_update_state_tree.circom

```diff
@@ -0,0 +1,124 @@
+include "./update_state_tree.circom";
+
+template BatchUpdateStateTree(
+  depth,
+  vote_options_tree_depth,
+  batch_size
+) {
+  // params:
+  //    depth: the depth of the state tree and the command tree
+  //    vote_options_tree_depth: depth of the vote tree
+  //    batch_size: that we're processing
+
+  // Output: New state tree root
+  signal output root;
+
+  // Input(s)
+  signal input coordinator_public_key[2];
+
+  var message_length = 11;
+  signal input message[batch_size][message_length];
+
+  // Select vote option index's weight
+  // (a.k.a the raw value of the leaf pre-hash)
+  signal private input vote_options_leaf_raw[batch_size];
+
+  // Vote options tree root (supplied by coordinator)
+  signal private input vote_options_tree_root[batch_size];
+  signal private input vote_options_tree_path_elements[batch_size][vote_options_tree_depth];
+  signal private input vote_options_tree_path_index[batch_size][vote_options_tree_depth];
+  signal input vote_options_max_leaf_index;
+
+  // Message tree
+  signal input msg_tree_root;
+  signal input msg_tree_path_elements[batch_size][depth];
+  signal input msg_tree_path_index[batch_size][depth];
+
+  // Random leaf (updated every batch)
+  signal private input random_leaf;
+  signal private input random_leaf_path_elements[depth];
+  signal private input random_leaf_path_index[depth];
```

Comment:
Isn't random leaf index always the last index? If yes, we can just use a locally calculated value instead of a signal (which might also cause some problems if you use an index that is different than the last one).

Reply:
fair point!